diff --git a/python-abp/abp/filters/parser.py b/python-abp_AdBlockID/abp/filters/parser.py
index 00f5661..1a982d5 100644
--- a/python-abp/abp/filters/parser.py
+++ b/python-abp_AdBlockID/abp/filters/parser.py
@@ -168,7 +168,6 @@ def _line_type(name, field_names, format_string):
 
 Header = _line_type('Header', 'version', '[{.version}]')
 EmptyLine = _line_type('EmptyLine', '', '')
-Comment = _line_type('Comment', 'text', '! {.text}')
 Metadata = _line_type('Metadata', 'key value', '! {0.key}: {0.value}')
 Filter = _line_type('Filter', 'text selector action options', '{.text}')
 Include = _line_type('Include', 'target', '%include {0.target}%')
@@ -337,7 +336,6 @@ def parse_line(line, position='body'):
             key, value = match.groups()
             if position != 'body' or key.lower() == 'checksum':
                 return Metadata(key, value)
-        return Comment(stripped[1:].lstrip())
 
     if stripped.startswith('%include') and stripped.endswith('%'):
         return _parse_instruction(stripped)
diff --git a/python-abp/abp/filters/renderer.py b/python-abp_AdBlockID/abp/filters/renderer.py
index dd7fcf1..8ac9e5a 100644
--- a/python-abp/abp/filters/renderer.py
+++ b/python-abp_AdBlockID/abp/filters/renderer.py
@@ -6,10 +6,10 @@ import itertools
 import logging
 import time
 
-from .parser import parse_filterlist, Comment, Metadata
+from .parser import parse_filterlist, Metadata
 from .sources import NotFound
 
-__all__ = ['IncludeError', 'MissingHeader', 'render_filterlist', 'render_diff']
+__all__ = ['IncludeError', 'MissingHeader', 'render_filterlist']
 
 _logger = logging.getLogger(__name__)
 
@@ -72,7 +72,7 @@ def _process_includes(sources, default_source, parent_include_stack, lines):
                     sources, inherited_source, include_stack, included)
 
                 _logger.info('- including: %s', name)
-                yield Comment('*** {} ***'.format(name))
+
                 for line in all_included:
                     if line.type not in {'header', 'metadata'}:
                         yield line
@@ -102,7 +102,28 @@ def _first_and_rest(iterable):
 def _insert_version(lines):
     """Insert metadata comment with version (a.k.a. date)."""
     first_line, rest = _first_and_rest(lines)
-    version = Metadata('Version', time.strftime('%Y%m%d%H%M', time.gmtime()))
+
+    # year.day_of_the_year.
+    # v_build = (datetime.datetime.utcnow().hour*60)+datetime.datetime.utcnow().minute
+    # version = Metadata('Version', time.strftime('%y.%j.{}'.format(v_build), time.gmtime()))
+
+    # year.month.number_of_commits_in_month
+    numberOfCommitsInMonth = subprocess.Popen(
+        [
+            'git', 'rev-list', 'HEAD', '--count', '--after="{} days"' '"+%Y-%m-%dT23:59"'
+            .format(datetime.datetime.now().day)
+        ],
+        stdout=subprocess.PIPE,
+        universal_newlines=True
+    )
+
+    version = Metadata(
+        'Version',
+        time.strftime('%y.X%m.{}'.format(numberOfCommitsInMonth.stdout.read().strip()),
+        time.gmtime()).replace('X0','X').replace('X','')
+        # https://stackoverflow.com/a/5900593
+    )
+
     return itertools.chain([first_line, version], rest)
 
 
@@ -158,8 +179,8 @@ def render_filterlist(name, sources, top_source=None):
     _logger.info('Rendering: %s', name)
     lines, default_source = _get_and_parse_fragment(name, sources, top_source)
     lines = _process_includes(sources, default_source, [name], lines)
-    for proc in [_process_timestamps, _insert_version, _remove_checksum,
-                 _validate]:
+    for proc in [_process_timestamps, _insert_version,
+                 _insert_checksum, _validate]:
         lines = proc(lines)
     return lines
 
